test/CouplerState/cplstate_interface.jl

```diff
@@ -38,7 +38,7 @@ end
 
     simA = SimulationA(ones(spaceA))
     simB = SimulationB(zeros(spaceB))
-    coupler = CouplerState()
+    coupler = CouplerState(1.0)
```

Comment:
> Coupled timestep storage: As the coupled timestepping info becomes more well defined, this may need to be refined (e.g. a CoupledTimestepper struct that could pair with the CouplerState struct as a full Coupler). This will likely be determined by how accumulation is ultimately handled.

This is one reason that having a separation here might be useful! You shouldn't really need time information when testing these `CouplerState` `put` and `get` methods. On the other hand, `push` and `pull` are more tied into the time coordination (since they would be part of a coupled step method). I still think we can punt on this.

Reply:
I agree with the separation between put/get and push/pull. But I would expect `Δt_coupled` to be in the `CoupledSimulation` to mimic the model simulations. There seems to be quite a bit of cross-over between the `CoupledSimulation` and `CoupledState`. Any thoughts on combining these?

Reply:
If we do this, it would be a separate PR anyway, so I'm happy to go ahead with merging this one.

Reply:
> I agree with the separation between put/get and push/pull. But I would expect `Δt_coupled` to be in the `CoupledSimulation` to mimic the model simulations. There seems to be quite a bit of cross-over between the `CoupledSimulation` and `CoupledState`. Any thoughts on combining these?

I think there is a separation - `CoupledSimulation` should fit the Simulations interface and deals with packing up component simulations _and_ the coupler (currently just a `CouplerState`) and deals with how to execute things. More on the time stepper side as a whole.

`CouplerState` deals with the actual coupled fields and passing them around. This is the part that will deal with MPI and how things get distributed.